api/application.go

```diff
@@ -114,7 +114,12 @@ func (h ApplicationHandler) Create(ctx *gin.Context) {
 		return
 	}
 	m := r.Model()
-	result := h.DB.Create(&m)
+	result := h.DB.Find(&model.Application{}, "name", m.Name)
```

Comment:
I have some concerns with this approach because it is vulnerable to race conditions where relying on the DB constraint is not.